core/adminserver/admin_server.go

```diff
@@ -213,11 +219,13 @@ func (s *Server) CreateDirectory(ctx context.Context, in *pb.CreateDirectoryRequ
 	// Generate VRF key.
 	wrapped, err := privKeyOrGen(ctx, in.GetVrfPrivateKey(), s.keygen)
 	if err != nil {
```

Comment:
could this just be:
if code := status.Code(err); code != codes.OK {
  return nil, status.Errorf(code, "adminserver: keygen(): %v", err)
}

Reply:
hmm, that would be better...